pyvisgen/simulation/scan.py

```diff
@@ -1,236 +1,197 @@
-    lm : 3d array
-        every pixel containing a l and m value
-    baselines : dataclass
-        baseline information
-    wave : float
-        wavelength of observation
-    time : astropy Time
-        Time steps of observation
-    src_crd : astropy SkyCoord
-        source position
-    array_layout : dataclass
-        station information
-    SI : 2d array
-        source brightness distribution / input img
+    rd : 3d tensor
+        every pixel containing ra and dec
+    ra : float
+        right ascension of source position
+    dec : float
+        declination of source position
 
     Returns
     -------
-    4d array
-        Returns visibility for every lm and baseline
+    2d array
+        Returns angular Distance for every pixel in rd grid with respect
+        to source position
     """
-    stat_num = array_layout.st_num.shape[0]
-    base_num = int(stat_num * (stat_num - 1) / 2)
+    r = rd[..., 0] - torch.deg2rad(ra.to(rd.device))
+    d = rd[..., 1] - torch.deg2rad(dec.to(rd.device))
+    theta = torch.arcsin(torch.sqrt(r**2 + d**2))
+    return theta
 
-    vectorized_num = np.vectorize(lambda st: st.st_num, otypes=[int])
-    st1, st2 = get_valid_baselines(baselines, base_num)
 
-    st1_num = vectorized_num(st1)
+@torch.compile
+def jinc(x):
+    """Create jinc function.
 
-    if st1_num.shape[0] == 0:
-        return torch.zeros(1)
+    Parameters
+    ----------
+    x : array
+        value of (?)
 
-    K = getK(baselines, lm, wave, base_num)
+    Returns
+    -------
+    array
+        value of jinc function at x
+    """
+    jinc = torch.ones(x.shape, device=x.device).double()
+    jinc[x != 0] = 2 * bessel_j1(x[x != 0]) / x[x != 0]
+    return jinc
 
-    B = np.zeros((lm.shape[0], lm.shape[1], 1), dtype=complex)
 
-    B[:, :, 0] = SI + SI
-    # # only calculate without polarization for the moment
-    # B[:, :, 0, 0] = SI[:, :, 0] + SI[:, :, 1]
-    # B[:, :, 0, 1] = SI[:, :, 2] + 1j * SI[:, :, 3]
-    # B[:, :, 1, 0] = SI[:, :, 2] - 1j * SI[:, :, 3]
-    # B[:, :, 1, 1] = SI[:, :, 0] - SI[:, :, 1]
+@torch.compile
+def integrate(X1, X2):
+    """Summation over (l,m) and avering over time and freq
 
-    X = torch.einsum("lmi,lmb->lmbi", torch.tensor(B), K)
+    Parameters
+    ----------
+    X1 : 3d tensor
+        visibility for every (l,m) and baseline for freq1
+    X2 : 3d tensor
+        visibility for every (l,m) and baseline for freq2
 
-    return X
+    Returns
+    -------
+    2d tensor
+    Returns visibility for every baseline
+    """
+    X_f = torch.stack((X1, X2))
+    int_m = torch.sum(X_f, dim=2)
+    del X_f
+    # only integrate for 1 sky dimension
+    # 2d sky is reshaped to 1d by sensitivity mask
+    # int_l = torch.sum(int_m, dim=2)
+    # del int_m
+    int_f = 0.5 * torch.sum(int_m, dim=0)
+    del int_m
+    X_t = torch.stack(torch.split(int_f, int(int_f.shape[0] / 2), dim=0))
+    del int_f
+    int_t = 0.5 * torch.sum(X_t, dim=0)
+    del X_t
+    return int_t
 
 
+'''
```

Comment:
Do we need the docstrings starting here or can we delete them?

Reply:
I deleted it